Graphics/Core/include/NovelRT/Graphics/GraphicsPushConstantRange.hpp

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include <NovelRT/Graphics/ShaderProgramVisibility.hpp>
+
+namespace NovelRT::Graphics
+{
+    struct GraphicsPushConstantRange
+    {
+        ShaderProgramVisibility visibilityFlags = ShaderProgramVisibility::NoneOrUnknown;
+        size_t offset = 0;
+        size_t size = 0;
```

Comment:
So we have to provide this data upfront for the pipeline, and then also provide it when calling the `PushConstants` command? More or less asking so I understand the reasoning behind these being stored here.

Reply:
Yes, its a metadata requirement when setting up the pipeline signature/layours. The CMD API I designed should handle the relevant data size assuming the buffer does not exceed the designated size that was initially specified here.

Reply:
https://registry.khronos.org/vulkan/specs/latest/man/html/vkCmdPushConstants.html for more information